src/app/services/error/error.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { PopUpError } from './../../models/pop-up-error.interface';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ErrorService {
+
+  errors: PopUpError[] = [];
+
+  addError(error: PopUpError): void {
+    this.errors.push(error);
+  }
+
+  getErrors(): PopUpError[] {
+    return this.errors;
+  }
+
+  clearError(error: PopUpError): void {
+    this.errors.splice(this.errors.indexOf(error), 1);
+  }
+
+  isErrors(): boolean {
+    return this.errors === [] ? false : true;
+  }
+
+  handleError(error: PopUpError): void {
+    error.handleFunction.call(error.callingObject);
```

Comment:
`handleFunction` is optional prop... please check if it exists before you use it.